File: leetCode/twoSum.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        HashMap<Integer, Integer> counter = new HashMap<Integer, Integer>();
        
        int j = 0; 
        while ( j < numbers.length) {
            if (numbers[j] > target) break;
            j++;
        };
        
        j = j == numbers.length ? numbers.length -1 : j;
        
        for (int i = 0; i <= j; i++) {
            if (counter.containsKey(target-numbers[i])) {
                int res[] = new int[] {counter.get(target-numbers[i]),i+1};
                return res;
            };
            counter.put(numbers[i],i+1);
        };
        
        return null;
    };
}

File: leetCode/twoSum.js
// two sum unsroted
var twoSum = function(numbers, target) {
    let counter = {};
    
    for (let i = 0; i < numbers.length; i++) {
        if (counter[target-numbers[i]]) return [counter[target-numbers[i]],i+1]; 
        counter[numbers[i]] = i+1;
    };
};

// if sorted burte forse
var twoSum = function(numbers, target) {
    let counter = {};
    let j = 0; 
    while (j < numbers.length && numbers[j] <= target) {
        j++;
    };
    
    for (let i = 0; i <= j; i++) {
        if (counter[target-numbers[i]]) return [counter[target-numbers[i]],i+1]; 
        counter[numbers[i]] = i+1;
    };
};

// sorted optimzed 
var twoSum = function(nums, target) {
    let first = 0; 
    let last = nums.length-1;
    
    while (first !== last) {
        if (nums[first] + nums[last] === target) return [first+1,last+1];
        if (nums[first] + nums[last] < target) {
            first++;
        } else {
            last--;
        };      
    };
};